zebra-network/src/protocol/external/codec/tests.rs

```diff
@@ -0,0 +1,366 @@
+
+    // i know the above msg has a body of 85 bytes
+    let size = 85;
+
+    // reducing the max size to body size - 1
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut bytes = Vec::new();
+        {
+            let mut fw = FramedWrite::new(
+                &mut bytes,
+                Codec::builder().with_max_body_len(size - 1).finish(),
+            );
+            fw.send(msg.clone())
+                .await
+                .expect_err("message should not encode as it is bigger than the max allowed value");
+        }
+    });
+
+    // send again with the msg body size as max size
+    let msg_bytes = zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut bytes = Vec::new();
+        {
+            let mut fw = FramedWrite::new(
+                &mut bytes,
+                Codec::builder().with_max_body_len(size).finish(),
+            );
+            fw.send(msg.clone())
+                .await
+                .expect("message should encode with the msg body size as max allowed value");
+        }
+        bytes
+    });
+
+    // receive with a reduced max size
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut fr = FramedRead::new(
+            Cursor::new(&msg_bytes),
+            Codec::builder().with_max_body_len(size - 1).finish(),
+        );
+        fr.next()
+            .await
+            .expect("a next message should be available")
+            .expect_err("message should not decode as it is bigger than the max allowed value")
+    });
+
+    // receive again with the tx size as max size
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut fr = FramedRead::new(
+            Cursor::new(&msg_bytes),
+            Codec::builder().with_max_body_len(size).finish(),
+        );
+        fr.next()
+            .await
+            .expect("a next message should be available")
+            .expect("message should decode with the msg body size as max allowed value")
+    });
+}
+
+/// Check that the version test vector deserializes correctly without the relay byte
+#[test]
+fn version_message_omitted_relay() {
+    let _init_guard = zebra_test::init();
+
+    let version = match VERSION_TEST_VECTOR.clone() {
+        Message::Version(mut version) => {
+            version.relay = false;
+            version.into()
+        }
+        _ => unreachable!("const is the Message::Version variant"),
+    };
+
+    let codec = Codec::builder().finish();
+    let mut bytes = BytesMut::new();
+
+    codec
+        .write_body(&version, &mut (&mut bytes).writer())
+        .expect("encoding should succeed");
+    bytes.truncate(bytes.len() - 1);
+
+    let relay = match codec.read_version(Cursor::new(&bytes)) {
+        Ok(Message::Version(VersionMessage { relay, .. })) => relay,
+        err => panic!("bytes should successfully decode to version message, got: {err:?}"),
+    };
+
+    assert!(relay, "relay should be true when omitted from message");
```

Comment:
Looks good!

Can you please add a test with a relay field/byte that is `false`, to catch bugs where we always set that field to `true`?

(I don't think this code has that bug, but we might refactor it in future.)